### PR TITLE
Re-attempt Playlist feature for Desktop on Nightly [Production]

### PR DESCRIPTION
**Pull request by bsclifton**
This is a redo of https://github.com/brave/brave-variations/pull/803

Contingent on review approval in https://github.com/brave/reviews/issues/1570

### CODE REVIEW

**Comment by kjozwiak** [comment]
Adding `Uplift/Blocked` until @bsclifton gives us the 👍 re: the security review passing as per https://github.com/brave/brave-variations/pull/997#issue-2234465240.

**Review by atuchin-m — APPROVED** [review]

**Comment by atuchin-m** [comment]
LGTM: https://github.com/brave/reviews/issues/1570#issuecomment-2057748989

**Review by kjozwiak — APPROVED** [review]
Uplift into `production` approved after deliberating with @brave/uplift-approvers. QA has verified the PR on `staging` as per https://github.com/brave/brave-variations/pull/996#issuecomment-2065150726,

**Pull request merged**
